x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -445,6 +445,10 @@ private SearchRequestBuilder createLatestDataCountsSearch(String indexName, Stri
                 .setIndicesOptions(IndicesOptions.lenientExpandOpen())
                 // look for both old and new formats
                 .setQuery(QueryBuilders.idsQuery().addIds(DataCounts.documentId(jobId), DataCounts.v54DocumentId(jobId)))
+                // We want to sort on log_time. However, this was added a long time later and before that we used to
+                // sort on latest_record_time. Thus we handle older data counts where no log_time exists and we fall back
+                // to the prior behaviour.
+                .addSort(SortBuilders.fieldSort(DataCounts.LOG_TIME.getPreferredName()).order(SortOrder.DESC).missing(0L))
```

Comment:
Do we have to worry about `unmapped type` errors?

Reply:
Definitely. I tricked myself thinking `missing` was doing the job `unmapped` needs to do. Will fix that, thanks for spotting it!

Reply:
Funny thing, I tested this. I created manually an index `.ml-anomalies-fake`. It all worked. But I didn't think that that index would pick our mappings due to the template being the new one.